ONSdigital/ssdc-rm-caseprocessor
# Integrate with RAS-RM for business sample load and case creation

# Motivation and Context
RAS-RM has all the respondent account features that RM is lacking, so we should find a way to make a mash-up of both products, to take advantage of all their features _combined_ into a single unified platform for all surveys that the ONS does, across all departments... business, social, health, census etc.

# What has changed
Proof of concept of "minimum viable" integration between RAS-RM and RM. At a minimum, RM should be able to load a business sample and process it to the point that Main Print Selection (MPS) can happen and the collection exercise can go live.

For this proof of concept, the sample is loaded into RM, then an action rule must be triggered to load the cases into RAS-RM, whereupon the normal MPS and go-live can happen in RAS-RM. Respondents and operational staff will continue to use RAS-RM Ops UI and respondents will continue to use Frontstage, as normal... but this is still the first serious integration between the products, and a nontrivial achievement.

# How to test?
You will need a RAS-RM sandbox. Start up your RAS-RM environment, then start RM up and load a business sample. Then set up a "Share with RAS-RM" action rule  (you'll need to add metadata like this: `{"rasRmCollectionExerciseId":"f41b3da6-0173-4e7d-bff0-70000f9f2b79","rasRmCollectionInstrumentId":"0f0f8c0f-7520-489c-a6e6-a00fa33a6b54"}`). Then the parties should be created in RAS-RM, and then the cases should be created in RAS-RM, and the MPS and go-live can happen like normal.

# Links
Trello: https://trello.com/c/zDenk9Gx
